packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -32,6 +32,12 @@ import {
 import { TRPCContextState } from '../../internals/context';
 import { getQueryKey } from '../../internals/getQueryKey';
 
+type DeepPartial<TObject> = TObject extends object
+  ? {
+      [P in keyof TObject]?: DeepPartial<TObject[P]>;
+    }
+  : TObject;
+
```

Comment:
Where would you put this utility type?

Reply:
I think we have most in trpc/server in some types.ts file

Reply:
Thanks! I moved the util type there